tutorials/AgileRL/agilerl_maddpg.py

```diff
@@ -154,16 +154,31 @@
     # Training loop
     for idx_epi in trange(max_episodes):
         for agent in pop:  # Loop through population
-            state = env.reset()[0]  # Reset environment at start of episode
+            state, info = env.reset()  # Reset environment at start of episode
             agent_reward = {agent_id: 0 for agent_id in env.agents}
             if INIT_HP["CHANNELS_LAST"]:
                 state = {
                     agent_id: np.moveaxis(np.expand_dims(s, 0), [3], [1])
                     for agent_id, s in state.items()
                 }
             for _ in range(max_steps):
-                action = agent.getAction(state, epsilon)  # Get next action from agent
-                next_state, reward, termination, truncation, _ = env.step(
+                agent_mask = info["agent_mask"] if "agent_mask" in info.keys() else None
+                env_defined_actions = (
+                    info["env_defined_actions"]
+                    if "env_defined_actions" in info.keys()
+                    else None
+                )
+
+                # Get next action from agent
+                cont_actions, discrete_action = agent.getAction(
```

Comment:
Why is it called cont actions plural but then discrete is only a single action? Seems confusing

Reply:
Because cont_actions is always in the shape of action dim, whereas discrete_action is simply the action that has been chosen to be taken. In this discrete case, cont_actions represents a one-hot encoded version of the action e.g. [0, 1, 0, 0] if the discrete action is 1, as this is needed for the RL algorithm to learn.